service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseInitiationService.java

```diff
@@ -187,10 +183,34 @@ public CaseData updateOrganisationsDetails(CaseData caseData) {
             return addCourtDetails(updatedCaseData);
         }
 
-        throw new IllegalStateException("Cannot determine local authority for a case");
+        if (nonNull(caseData.getRepresentativeType())) {
+            boolean isRespondentSolicitor = Objects.equals(caseData.getRepresentativeType().toString(),
+                "RESPONDENT_SOLICITOR");
+            boolean isChildSolicitor = Objects.equals(caseData.getRepresentativeType().toString(),
+                "CHILD_SOLICITOR");
+
+            if (isRespondentSolicitor || isChildSolicitor) {
+                CaseData updatedCaseData = caseData.toBuilder()
+                    .outsourcingPolicy(organisationPolicy(
+                        currentUserOrganisationId,
+                        currentUserOrganisationName,
+                        isRespondentSolicitor ? SOLICITORA : CHILDSOLICITORA))
+                    .build();
+
+                return addCourtDetails(updatedCaseData);
+            }
+        }
```

Comment:
What should happen if the type is local authority? Do we still need the exception for that case?